src/blas/backends/mklgpu/mklgpu_batch.cxx

```diff
@@ -19,160 +19,281 @@
-    ::oneapi::mkl::gpu::dgemm_batch(queue, MAJOR, ::mkl::cblas_convert(transa),
-                                    ::mkl::cblas_convert(transb), m, n, k, alpha, a, lda, stride_a,
-                                    b, ldb, stride_b, beta, c, ldc, stride_c, batch_size);
-}
-
-void gemm_batch(cl::sycl::queue &queue, oneapi::mkl::transpose transa,
-                oneapi::mkl::transpose transb, std::int64_t m, std::int64_t n, std::int64_t k,
-                std::complex<float> alpha, cl::sycl::buffer<std::complex<float>, 1> &a,
-                std::int64_t lda, std::int64_t stride_a,
-                cl::sycl::buffer<std::complex<float>, 1> &b, std::int64_t ldb,
-                std::int64_t stride_b, std::complex<float> beta,
-                cl::sycl::buffer<std::complex<float>, 1> &c, std::int64_t ldc,
-                std::int64_t stride_c, std::int64_t batch_size) {
-    ::oneapi::mkl::gpu::cgemm_batch(queue, MAJOR, ::mkl::cblas_convert(transa),
-                                    ::mkl::cblas_convert(transb), m, n, k, alpha, a, lda, stride_a,
-                                    b, ldb, stride_b, beta, c, ldc, stride_c, batch_size);
-}
-
-void gemm_batch(cl::sycl::queue &queue, oneapi::mkl::transpose transa,
-                oneapi::mkl::transpose transb, std::int64_t m, std::int64_t n, std::int64_t k,
+void copy_batch(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<float, 1> &x, int64_t incx,
+                std::int64_t stridex, cl::sycl::buffer<float, 1> &y, int64_t incy,
+                std::int64_t stridey, std::int64_t batch_size) {
+    ::oneapi::mkl::gpu::scopy_batch_sycl(&queue, n, &x, incx, stridex, &y, incy, stridey,
+                                         batch_size);
+}
+
+void copy_batch(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<double, 1> &x, int64_t incx,
+                std::int64_t stridex, cl::sycl::buffer<double, 1> &y, int64_t incy,
+                std::int64_t stridey, std::int64_t batch_size) {
+    ::oneapi::mkl::gpu::dcopy_batch_sycl(&queue, n, &x, incx, stridex, &y, incy, stridey,
+                                         batch_size);
+}
+
+void copy_batch(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<std::complex<float>, 1> &x,
+                int64_t incx, std::int64_t stridex, cl::sycl::buffer<std::complex<float>, 1> &y,
+                int64_t incy, std::int64_t stridey, std::int64_t batch_size) {
+    ::oneapi::mkl::gpu::ccopy_batch_sycl(&queue, n, &x, incx, stridex, &y, incy, stridey,
+                                         batch_size);
+}
+
+void copy_batch(cl::sycl::queue &queue, int64_t n, cl::sycl::buffer<std::complex<double>, 1> &x,
+                int64_t incx, std::int64_t stridex, cl::sycl::buffer<std::complex<double>, 1> &y,
+                int64_t incy, std::int64_t stridey, std::int64_t batch_size) {
+    ::oneapi::mkl::gpu::zcopy_batch_sycl(&queue, n, &x, incx, stridex, &y, incy, stridey,
```

Comment:
Right, that is my preference as well due to same reasons you mentioned that I had to work around before. We had some discussions about this before but did not reach to a conclusion. One concern with this could be having the exact same function signature for oneMKL and oneMKL open source interfaces at runtime. 
@mkrainiuk Do you have more info about this discussion?

Nevertheless, this change should be done as a separate PR since it impact all existing APIs as well.

Reply:
Right, the main problem with using Intel oneMKL external API is a potential recursion, because opensource wrapper can resolve call to Intel oneMKL as opensource oneMKL symbol again and go to the same wrapper.

Reply:
Is there some tracker open for this?

Reply:
Thanks for reminding. Opened an issue: #114